src/category_theory/closed/ideal.lean

```diff
@@ -0,0 +1,278 @@
+/-
+Copyright (c) 2021 Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bhavik Mehta
+-/
+import category_theory.limits.preserves.shapes.binary_products
+import category_theory.limits.shapes.constructions.finite_products_of_binary_products
+import category_theory.monad.limits
+import category_theory.adjunction.fully_faithful
+import category_theory.adjunction.reflective
+import category_theory.closed.cartesian
+import category_theory.subterminal
+
+/-!
+# Exponential ideals
+
+An exponential ideal of a cartesian closed category `C` is a subcategory `D ⊆ C` if for any `B : D`
+and `A : C`, the exponential `B^^A` is in `D` - resembling ring theoretic ideals. We define the
```

Comment:
```suggestion
and `A : C`, the exponential `B^^A` is in `D` --- resembling ring theoretic ideals. We define the
```

Reply:
Amended, thanks! I'm not sure what you meant by changing the `-` to a `---`; perhaps a doc-gen thing I'm not aware of? In any case I've changed to a colon, hopefully that's fine?